app/models/course.rb

```diff
@@ -17,6 +17,16 @@ class Course < ActiveRecord::Base
               message: 'should not contain white spaces'
             }
 
+  validates :title,
+            presence: true,
+            length: { within: 1..40 }
+
+  validates :material_url,
+            format: {
+                without: /:/,
+                message: 'should not contain colon (:). Remove "http://" from the beginning if you have it'
```

Comment:
But what about https URLs?

I'd much rather require `/^https?:\/\//` in this field.